server/servercontext.py

```diff
@@ -53,24 +54,59 @@ def close(self):
     def __contains__(self, connection):
         return connection in self.connections.keys()
 
+    async def broadcast(self, message, validate_fn=lambda a: True):
+        await self._do_broadcast(
+            validate_fn,
+            QDataStreamProtocol.send_message,
+            message
+        )
+        self._logger.log(TRACE, "]]: %s", message)
+
     async def broadcast_raw(self, message, validate_fn=lambda a: True):
```

Comment:
do we even need that function? It's used only for `ping` every 45 secs

Reply:
Yes, we need it for precisely this reason. This is also why I refactored the main logic into `_do_broadcast` so that I this function can be a single function call.